### PR TITLE
add python, rest api test

### PR DESCRIPTION
**Pull request by Kartik14**
This PR adds tests for the Python and Rest api for few different cases. Currently It uses models `Llama-2-7b-chat-hf-q4f16_1` present in the `dist` directory, which would probably need to be changed for Github testing.

### CODE REVIEW

File: tests/python/api/test_python.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+def test_chat_module_creation_and_generate_with_stream(model: str):
+    chat_module = ChatModule(model=model)
+    _ = chat_module.generate(
+        prompt="How to make a cake?",
+        progress_callback=StreamToStdout(callback_interval=2),
+    )
+    print(f"Statistics: {chat_module.stats()}\n")
+
+
+@pytest.mark.parametrize(
```
Comment (davidpissarra) [inline tests/python/api/test_python.py:29]:
```
@pytest.mark.parametrize("generation_config, model", ...)

**Review by davidpissarra — APPROVED** [review]
Thanks @Kartik14 lgtm, please fix the lint issue, and these two decorators

File: tests/python/api/test_rest.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                break
+            except requests.exceptions.ConnectionError:
+                time.sleep(1)
+        yield
+        server_proc.send_signal(signal.SIGINT)
+        server_proc.wait()
+
+
+@pytest.mark.usefixtures("run_rest_server")
+@pytest.mark.parametrize("stream", [True, False])
```
[the remark this answers is not in the document]
Comment (Kartik14) [reply tests/python/api/test_rest.py:33]: Thanks for review @davidpissarra. I am not sure the above change would work. Instead I would have to do something like this
```
@pytest.mark.parametrize(
    "stream, model", [(stream, model) for stream in [True, False] for model in MODELS]
)
```
I think having multiple pytest.mark.parameterize is cleaner than the above code. But I am not sure what the standard practice is.. What do you think?

Also, the pylint error is due to `request` package not installed in the docker image `mlcaidev/ci-cpu` used for ci testing. Do you know where the dockerfile for that image is?

[the remark this answers is not in the document]
Comment (davidpissarra) [reply tests/python/api/test_rest.py:33]: You can install the `requests` package in the pylint ci .sh https://github.com/mlc-ai/mlc-llm/blob/main/ci/task/pylint.sh

[the remark this answers is not in the document]
Comment (davidpissarra) [reply tests/python/api/test_rest.py:33]: > I think having multiple pytest.mark.parameterize is cleaner than the above code. But I am not sure what the standard practice is.. What do you think?

Sounds good, we can keep this one like that, I thought there was a cleaner way to have just one parametrize call

**Review by davidpissarra — APPROVED** [review]

**Pull request merged**